src/components/Navigation/PageMenu/PageMenu.tsx

```diff
@@ -1,29 +1,92 @@
-import { FC } from 'react';
+import { FC, useMemo } from 'react';
 
+import { tokens } from '@equinor/eds-tokens';
+
+import { PageMenuVariants } from './PageMenu.types';
 import PageMenuItem from './PageMenuItem';
+import { BorderItemsContainer } from './PageMenuItem.styles';
 import { usePageMenu } from 'src/providers/PageMenuProvider';
+import { getValues } from 'src/providers/PageMenuProvider.utils';
 import { spacings } from 'src/style';
 
 import styled from 'styled-components';
 
-const Container = styled.div`
+const { colors } = tokens;
+
+interface ContainerProps {
+  $variant: PageMenuVariants;
+}
+
+const Container = styled.div<ContainerProps>`
   display: flex;
   flex-direction: column;
-  gap: ${spacings.small};
+  ${({ $variant }) => {
+    switch ($variant) {
+      case 'buttons':
+        return `gap: ${spacings.small};`;
+      case 'border':
+        return `border-left: 2px solid ${colors.ui.background__medium.rgba};`;
+    }
+  }}
 `;
 
-interface PageMenuProps {
+export interface PageMenuProps {
```

Comment:
I don't know if the name PageMenu reflects what this component is called in Figma, but I don't think it's a very good name, a bit too generic maybe. Could it be called TableOfContents instead?

Reply:
Yeah agree PageMenu is a poor name, I see it is called TOC in PWEX figma and Side navigation in the portal figma 😅 IMO the name you suggested makes the most sense so I'll change it to that